### PR TITLE
doc/architecture: edit "SDEH"

### PR DESCRIPTION
**Pull request by zdover23**
Edit the front matter of the "Smart Daemons Enable Hyperscale" section of doc/architecture.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/architecture.rst [hunk 1]
```diff
@@ -375,22 +375,25 @@ tickets and session keys.
...
-Daemons in the cluster.  This enables Ceph OSD Daemons to interact directly with
-other Ceph OSD Daemons and Ceph Monitors. Additionally, it enables Ceph Clients
-to interact directly with Ceph OSD Daemons.
-
-The ability of Ceph Clients, Ceph Monitors and Ceph OSD Daemons to interact with
-each other means that Ceph OSD Daemons can utilize the CPU and RAM of the Ceph
-nodes to easily perform tasks that would bog down a centralized server. The
-ability to leverage this computing power leads to several major benefits:
+A feature of many storage clusters is a centralized interface that keeps track
+of the nodes that the clients are permitted to access. Such centralized
```
Comment (anthonyeleven) [inline doc/architecture.rst:379]: that clients

File: doc/architecture.rst [hunk 2]
```diff
@@ -375,22 +375,25 @@ tickets and session keys.
...
+architectures provide services to clients by means of a double dispatch. At the
+petabyte-to-exabyte scale, such double dispatches generate a significant
```
Comment (anthonyeleven) [inline doc/architecture.rst:381]: s/generate/are/

File: doc/architecture.rst [hunk 3]
```diff
@@ -375,22 +375,25 @@ tickets and session keys.
...
+bottleneck.
+
+Ceph eliminates this bottleneck: Ceph's OSD Daemons AND Ceph clients are
```
Comment (anthonyeleven) [inline doc/architecture.rst:384]: I want to say obviates, but that might violate your guidance on being too outré for ESOL readers

File: doc/architecture.rst [hunk 4]
```diff
@@ -375,22 +375,25 @@ tickets and session keys.
...
+cluster aware. Like Ceph clients, each Ceph OSD Daemon is aware of other Ceph
```
Comment (anthonyeleven) [inline doc/architecture.rst:385]: I really want to hypenate that.

File: doc/architecture.rst [hunk 5]
```diff
@@ -375,22 +375,25 @@ tickets and session keys.
...
+OSD Daemons in the cluster. This enables Ceph OSD Daemons to interact directly
+with other Ceph OSD Daemons and to interact directly with Ceph Monitors.  Being
+cluster-aware makes it possible for Ceph clients to interact directly with Ceph
```
Comment (anthonyeleven) [inline doc/architecture.rst:388]: hah!

File: doc/architecture.rst [hunk 6]
```diff
@@ -375,22 +375,25 @@ tickets and session keys.
...
+OSD Daemons.
+
+Because Ceph clients, Ceph monitors, and Ceph OSD daemons can interact with one
```
Comment (anthonyeleven) [inline doc/architecture.rst:391]: s/can//

This convergence on repeated _Ceph xxx_ is a bit awkward but I can understand the desire for uniform nomenclature.

File: doc/architecture.rst [hunk 7]
```diff
@@ -375,22 +375,25 @@ tickets and session keys.
...
+another directly, Ceph OSD daemons can make use of the CPU and the RAM of the
```
Comment (anthonyeleven) [inline doc/architecture.rst:392]: use of the aggregate CPU and RAM resources of the

**Review by anthonyeleven — APPROVED** [review]

File: doc/architecture.rst (hunk 3 above)
Comment (zdover23) [reply doc/architecture.rst:384]: I like "obviates". You're the one who but the kibosh on it before.

We're going with "obviates".

File: doc/architecture.rst (hunk 1 above)
Comment (zdover23) [reply doc/architecture.rst:379]: Accepted.

File: doc/architecture.rst (hunk 2 above)
Comment (zdover23) [reply doc/architecture.rst:381]: Accepted.

File: doc/architecture.rst (hunk 3 above)
Comment (zdover23) [reply doc/architecture.rst:384]: Accepted.

File: doc/architecture.rst (hunk 4 above)
Comment (zdover23) [reply doc/architecture.rst:385]: It should have been hyphenated in the first instance. 

Now it is.

File: doc/architecture.rst (hunk 6 above)
Comment (zdover23) [reply doc/architecture.rst:391]: "can" was a just stopgap measure that helped me with the cognitive load of recasting of this paragraph, and you're right. There's no reason to be modal here. Plain old indicative is better.

Comment (zdover23) [reply doc/architecture.rst:391]: And I envision the reader of the architecture document to be someone who has just finished reading the not-yet-available Beginner's Guide, and who isn't completely sure if a monitor is a Ceph monitor and so on, and so needs to be reminded.

File: doc/architecture.rst (hunk 7 above)
Comment (zdover23) [reply doc/architecture.rst:392]: Yeah, that is better.

Accepted.

File: doc/architecture.rst (hunk 6 above)
Comment (anthonyeleven) [reply doc/architecture.rst]: Ack.  Plus it's very common for someone to use a search engine to find our works without starting at the beginning.

There was this lawyer on a plane once telling his daughter over the phone about how he bought the regular size candy bar then bought the large size candy bar as well as the regular size candy bar.  Stuff like that I can't unhear.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/53659 - Reef backport
https://github.com/ceph/ceph/pull/53660 - Quincy backport

**Pull request merged**